internal/rbd/nodeserver.go

```diff
@@ -294,6 +295,13 @@ func (ns *NodeServer) stageTransaction(ctx context.Context, req *csi.NodeStageVo
 	util.DebugLog(ctx, "rbd image: %s/%s was successfully mapped at %s\n",
 		req.GetVolumeId(), volOptions.Pool, devicePath)
 
+	if volOptions.Mounter == rbdNbdMounter {
```

Comment:
Actually, we can get the path for krbd too, but since we do not need the path of krbd device (/dev/rbdX) for later use, this check will save some cycles.

I'm fine to update the device for krbd too, but IMHO that is of no use.

I'm happy to hear your further thoughts.